app/scripts/modules/core/src/pipeline/config/triggers/artifacts/helm/HelmArtifactEditor.tsx

```diff
@@ -80,11 +107,21 @@ class HelmEditor extends React.Component<IArtifactEditorProps, IHelmArtifactEdit
     this.props.onChange(clone);
   };
 
-  private onNameChange = (chartName: string) => {
-    ArtifactService.getArtifactVersions(TYPE, this.props.account.name, chartName).then(versions => {
-      this.setState({ versions });
+  private getChartVersionOptions(chartName: string) {
+    const { artifact, account } = this.props;
+    this.setState({ versionsLoading: true });
+    ArtifactService.getArtifactVersions(TYPE, account.name, chartName).then((versions: string[]) => {
+      // if the version doesn't match SEMVER we assume that it's a regular expression or SpEL expression
+      // and add it to the list of valid versions
+      if (artifact.version && !new RegExp(SEMVER).test(artifact.version)) {
```

Comment:
Might possibly be cleaner to create the RegExp only once as a private member of this class?